pkg/migrator/core_test.go

```diff
@@ -235,3 +240,68 @@ func TestMigrateListClusterScoped(t *testing.T) {
 		}
 	}
 }
+
+type fakeProgress struct{}
+
+func (f *fakeProgress) load() (string, error) {
+	return "", nil
+}
+
+func (f *fakeProgress) save(string) error {
+	return nil
+}
+
+func TestMetrics(t *testing.T) {
+	metrics.Metrics.Reset()
+	// fake client doesn't support pagination, so we can't test complex behavior.
+	nodeList := newNodeList(100)
+	client := fake.NewSimpleDynamicClient(scheme.Scheme, toUnstructuredListOrDie(nodeList))
+	migrator := NewMigrator(v1.SchemeGroupVersion.WithResource("nodes"), client, &fakeProgress{})
+	migrator.Run()
+	expectCounterCount(t,
+		"storage_migrator_core_migrator_migrated_objects",
+		map[string]string{
+			"resource": "/v1, Resource=nodes",
+		},
+		100,
+	)
+
+}
+
+func labelsMatch(metric *ptype.Metric, labelFilter map[string]string) bool {
+	for _, lp := range metric.GetLabel() {
+		if value, ok := labelFilter[lp.GetName()]; ok && lp.GetValue() != value {
+			return false
+		}
+	}
+	return true
+}
+
+func expectCounterCount(t *testing.T, name string, labelFilter map[string]string, wantCount int) {
+	metrics, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		t.Fatalf("Failed to gather metrics: %s", err)
+	}
+	counterSum := 0
+	for _, mf := range metrics {
+		if mf.GetName() != name {
+			continue // Ignore other metrics.
+		}
+		for _, metric := range mf.GetMetric() {
+			if !labelsMatch(metric, labelFilter) {
+				continue
+			}
+			counterSum += int(metric.GetCounter().GetValue())
+		}
+	}
+	if wantCount != counterSum {
+		t.Errorf("Wanted count %d, got %d for metric %s with labels %#+v", wantCount, counterSum, name, labelFilter)
+		for _, mf := range metrics {
```

Comment:
Is there some reason we need this extra logging? It looks like the metric information should already be there and we already know the want/got count.

Reply:
This helps debugging in case of test failure. For example, if my code uses the wrong labels, the t.Logf prints out all the metrics so that I can see all the wrong labels.